# Axe 3 — Satisfaction & implication

Dans ce notebook, j'étudie les variables liées aux facteurs subjectifs des employés (le ressenti et l'investissement) afin d'évaluer leur impact sur l'attrition.

Ce notebook a pour objectif :
- d'explorer les variables de l'axe,
- de comparer avec l'attrition
- d'explorer des relations internes avec les autres axe 
- de visualiser
- retenir seulement ce qui ressort clairement
---
**Remarque a moi**
Pourquoi, à conditions égales, certains partent et d’autres non ?

## 1. Exploration desciptive des variables de l’axe
---

### 1.1 Import des librairies & chargement des données
---

In [1]:
# import librairies 

import pandas as pd
import pyarrow
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import os

In [2]:
# chargement du df
df_clean = pd.read_parquet("C:/Users/Kemu/Documents/Formation/Projet Pro Data/2_Projets_Tests/projet-RH_test/data/processed/employees_clean.parquet")
df_clean.columns

Index(['Age', 'Attrition', 'BusinessTravel', 'DailyRate', 'Department',
       'DistanceFromHome', 'Education', 'EducationField', 'EmployeeCount',
       'EmployeeNumber', 'EnvironmentSatisfaction', 'Gender', 'HourlyRate',
       'JobInvolvement', 'JobLevel', 'JobRole', 'JobSatisfaction',
       'MaritalStatus', 'MonthlyIncome', 'MonthlyRate', 'NumCompaniesWorked',
       'Over18', 'OverTime', 'PercentSalaryHike', 'PerformanceRating',
       'RelationshipSatisfaction', 'StandardHours', 'StockOptionLevel',
       'TotalWorkingYears', 'TrainingTimesLastYear', 'WorkLifeBalance',
       'YearsAtCompany', 'YearsInCurrentRole', 'YearsSinceLastPromotion',
       'YearsWithCurrManager'],
      dtype='object')

### 1.2 Fonctions utilitaires

In [ ]:
# Tableau de pourcentage variable univarié

def value_counts_percent(serie,round_n=3):

    """  Calcule la distribution (%) d'une variable catégorielle et retourne un DataFrame"""
    
    return (serie.value_counts(normalize=True).
            round(round_n).
            mul(100).
            reset_index(name="Pourcentage"))

# Tableau de pourcentage de variables bivariés
def groupby_percent(df,columns1,columns2,observed=True):
    """Calcule la distribution (%) de columns2 pour chaque modalité de columns1
    et retourne un DataFrame au format long. """
    return(df.groupby(columns1,observed=observed)[columns2].
           value_counts(normalize=True).
           round(2)
           *100).reset_index(name="Pourcentage")

# Table pivot en pourcentage de variables bivarié
def pivot_percent(df,index,columns):
    """ Crée un tableau pivot (%) : lignes=index, colonnes=columns."""
    df = groupby_percent(df,index,columns)
    return(df.pivot(index=index, columns=columns,values="Pourcentage").fillna(0))




### 1.3 Creation du DataFrame satisfaction_implication
---
#### Colonnes incluses dans l'axe "Satisfaction et implication"

Cet axe regroupe les caractéristiques du ressenti et de l'investissement des employés dans la structure:
- JobInvolvement = implication
- EnvironmentSatisfaction = satisfaction
- JobSatisfaction = satifaction
- PerformanceRating = implication (à manier prudemment)
- RelationshipSatisfaction = satisfaction
- WorkLifeBalance = satisfaction


In [3]:
# colonnes condition de travail
cols_satisfaction_implication = ['JobInvolvement', 'EnvironmentSatisfaction','JobSatisfaction',"PerformanceRating",'RelationshipSatisfaction','WorkLifeBalance']

# creation df_condition_Trav
df_satisfaction_implication = df_clean[cols_satisfaction_implication + ['Attrition']].copy()
df_satisfaction_implication.head(3)

,JobInvolvement,EnvironmentSatisfaction,JobSatisfaction,PerformanceRating,RelationshipSatisfaction,WorkLifeBalance,Attrition
0,3,2,4,3,1,1,Yes
1,2,3,2,4,4,3,No
2,2,4,3,3,2,3,Yes


#### 1.3.1 Préparation des variables
- Renommage pour la lisibilité
- Création de tranches
- Regroupements :Les seuils de regroupement peuvent différer selon les sous-thèmes afin de s’adapter à la distribution spécifique de chaque variable et à leur interprétation métier.


##### Renommage 



##### Création des tranches 



### 1.4  Analyse individuelle des variables 
---
Dans cette section, l’analyse porte sur plusieurs caractéristiques de la satisfaction et de l'implication des employés, afin de mieux comprendre le contexte organisationnel de la population étudiée.

Les variables sont analysées selon deux sous-thèmes :
- la satisfaction :  perception et ressenti (environnement, relations, équilibre)
- l'implication : comportement et investissement

Certaines variables seront analysées de manière plus approfondie que d’autres.
Cette analyse repose sur une approche **descriptive**, basée sur l’étude des **distributions**, à l’aide de **tableaux de proportions** et de **visualisations graphiques**.

**Remarque moi**
   - Présentation des variables
   - Distribution globale des niveaux de satisfaction et d’implication
   - Lecture synthétique

#### 1.4.1 La satisfaction
---
Cette section est consacrée à l’exploration des variables liées à la **la satisfaction**, qui regroupent les caractéristiques organisationnelles du travail et certaines conditions professionnelles des employés.

Les variables étudiées sont les suivantes :

- **EnvironmentSatisfaction** : niveau de satisfaction vis-à-vis de l’environnement et des conditions de travail  
- **JobSatisfaction** : niveau global de satisfaction au travail  
- **RelationshipSatisfaction** : niveau de satisfaction des relations professionnelles (collègues, manager)  
- **WorkLifeBalance** : niveau de satisfaction concernant l’équilibre entre vie professionnelle et vie personnelle


Dans le cadre de cette analyse :
- **x** feront l’objet d’une analyse descriptive légère.
-  **y** seront analysées de manière plus détaillée.


Les variables de ce sous-thème sont décrites à travers leurs répartitions afin de mieux comprendre la structure des postes au sein de l’entreprise.

##### 1.4.1.1 

##### Tableau des proportions

***Observation descriptive***

##### 1.4.1.2 

##### Tableau des proportions

##### Répartitions des employés selon leur fonction

***Observation descriptive***

##### 1.4.1.3 

##### Tableau des proportions

***Observation descriptive***

##### 1.4.1.4 

##### Tableau des proportions

***Observation descriptive*** ou **Visuel**

##### Répartitions des employés en fonctions

***Observation descriptive***

**Mini-synthèse - Satisfaction**



#### 1.4.2 Implication
---
Cette section s’intéresse aux variables liées à la **l'inplication et l'investissement** des employes.

Les variables analysées sont :

- **JobInvolvement** : niveau d’engagement professionnel de l’employé dans son travail  
- **PerformanceRating** : évaluation du niveau de performance observée  
  *(proxy indirect de l’engagement, à interpréter avec prudence)* 


Dans le cadre de cette analyse :
- **x** fera l’objet d’une analyse descriptive légère.
- **y***  sera analysées de manière plus détaillée.

 
Ces variables sont analysées de manière descriptive afin de caractériser la distribution des niveaux de rémunération au sein de l’entreprise.

##### 1.4.2.1 

##### Tableau des proportions

##### Répartitions des employés en fonctions de 

***Observation descriptive***


##### 1.4.2.2 

##### Tableau des proportions

***Observation descriptive***

**Mini-synthèse - implication**


Après avoir décrit les principales caractéristiques de la satification et implication des employés, cette section s’intéresse désormais à leur relation avec l’attrition.

L’objectif est d’observer si certaines dimensions subjective  telles que la satisfaction et l'implication présentent des différences de taux d’attrition selon les groupes étudiés.


## 2. Attrition - Satifcation/Implication
---

### 2.1 Introduction
---
Dans cette section, je compare la variable cible, l’attrition, aux variables des conditions de travail analysées précédemment, afin d’identifier d’éventuelles relations entre ces caractéristiques et le départ des employés.

##### Tableau de proportions Attrition Yes / No

In [ ]:
attrition = df_satisfaction_implication['Attrition'].value_counts(normalize=True).sort_values(ascending=False).round(3)*100

df_attrition = attrition.reset_index()
df_attrition.columns = ['Attrition', "Pourcentage"]

df_attrition

,Attrition,Pourcentage
0,No,83.9
1,Yes,16.1


D’après ce tableau, le taux d’attrition global observé dans le jeu de données est d’environ **16 %**, ce qui constitue le niveau de référence pour les analyses suivantes.
Dans les analyses suivantes, l’accent est mis sur le taux d’attrition (Attrition = Yes), les non-départs représentant le complément à 100 %.

### 2.2 Analyse de l’attrition selon les variables de la satisfaction et l'implication
---
Dans cet partie, les sous themes précedent sont réutilisé pour analyser les variables avec l'attrition.

Certaines variables de cet axes ont été explorées individuellement mais n’ont pas été retenues dans l’analyse de l’attrition, en raison de distributions peu contrastées ou d’un intérêt analytique limité observé lors de l’exploration initiale.

Les variables sélectionnées sont comparées à partir de leur fréquence, à l’aide de tableaux de proportions et de visualisations graphiques.  
Cette approche permet d’identifier des différences de taux d’attrition entre les groupes avant toute analyse approfondie.

#### 2.2.1 Attrition et la satisfaction
---

Dans cette partie, les variables analysé avec l'attrition sont  


##### 2.2.1.1 Attrition et 
---

##### Tableau de proportions ­­

##### Graphique — Analyse de la relation entre le poste occupé et l'attrition

***Observation descriptive***

#### 2.2.1.2 Attrition et 
---

##### Tableau de proportions ­­

##### Graphique — Analyse de la relation entre les heures supplémentaires et l'attrition

***Observation descriptive***

#### 2.2.2 Attrition et l'implication
---

Dans cette partie, les variables analysé avec l'attrition sont 


#### 2.2.2.1 Attrition et 
---

##### Tableau de proportions ­­

##### Graphique — Analyse de la relation entre le revenu mensuel et l'attrition

***Observation descriptive***

#### 2.2.2.2 Attrition et 
---

##### Tableau de proportions ­­

***Observation descriptive***

#### 2.3 Résultats clés - Satisfaction /implication et attrition
---


Une fois les comparaisons simples réalisées, j’explore les facteurs explicatif à l’axe afin d’identifier d’éventuels patterns ou corrélations.

## 3. Facteurs associés à la satisfaction et à l’implication
----

### 3.1 Introduction

Cette section vise à identifier les facteurs organisationnels ou individuels associés aux niveaux de satisfaction et d’implication, afin de mieux contextualiser les résultats observés sur l’attrition.

l'analyse portera sur trois ou quatre relation, qui m'ont semblé pertinent d'etudier afin de donner du contexte au résultat de l'attriton deja observé :

- 
- 
- 
---
**Remarque moi**
Satisfaction × quelques variables clés de l’axe 2
Implication × quelques variables clés de l’axe 2
Éventuellement 1 variable perso

### 3.2 Analyse des relations entre Satisfaction/ implication et les variables organisationnelles
---

Les variables sont analysées à partir de tableaux croisés (%) et de visualisations graphiques descriptives.

#### 3.2.1 Relation entre 
---

##### Tableaux croisés (%)

##### Visualisation


***Observation descriptive***

#### 3.2.2 Relation entre 
---


##### Tableaux croisés (%)

##### Visualisation

***Observation descriptive***

#### 3.2.3 Relation 
---


##### Tableau croisé (%)

##### Visualisation



***Observation descriptive***

### 3.3 Analyse des relations entre Satisfaction/ implication et les variables individuelle
---

Les variables sont analysées à partir de tableaux croisés (%) et de visualisations graphiques descriptives.

#### 3.2.1 Relation entre 
---

##### Tableaux croisés (%)

##### Visualisation


***Observation descriptive***

Après avoir analysé ces relations, cette section synthétise les principaux patterns observés.

### 3.4 Synthèse des Patterns transversaux
---



# 4. Résultats clés 
---


---


# 5. Conclusion de l’axe “Condition de travail”
---
